src/bsoncxx/view_or_value.hpp

```diff
@@ -118,7 +118,7 @@ class view_or_value {
         return _view;
     }
 
-   private:
+   protected:
```

Comment:
I see why you did this so you can access them to implement c_str above, but I'm not sure you actually need to. I'd prefer to keep these private.